tests/snapshot_tests.rs

```diff
@@ -0,0 +1,75 @@
+//! Snapshot tests of `cargo semver-checks` runs to ensure
```

Comment:
Could we get away with keeping this inside a `#[cfg(test)]` within the crate itself, instead of as a separate module inside the top-level `tests`?

There's a substantial perf and ergonomics difference between the two: the former get compiled as part of the crate itself whereas the latter get compiled as a separate crate and are much slower to run and iterate on in my experience.

Reply:
By "this" do you mean the tests themselves or just the infrastructure for snapshot testing?

Reply:
The whole thing. If possible, ideally we wouldn't add anything to the top-level `tests/` at all.

Reply:
I can try but I think it depends on having the `cargo-semver-checks` binary already built, so I don't know if it would work.

Reply:
If we could attach to "the layer underneath" then the only thing we lose out on testing is making sure clap correctly parsed our arguments. That doesn't seem _that_ valuable by comparison, especially if we get much better test perf and iteration speed.

If you don't think it's possible, I think this is fine to merge too. But it's worth a look at whether it's possible, just because the iteration speed would be different by multiples, not percentages.

Reply:
I think I was wrong - it seems to work anyway, so I moved it to lib tests.  Not sure if it will still be slower because it needs to build the binary? but if so I can do like you suggest.

Reply:
This seems highly suspect.

`insta-cmd`'s [`get_cargo_bin` implementation](https://github.com/mitsuhiko/insta-cmd/blob/main/src/cargo.rs#L68-L73) uses [the `CARGO_BIN_EXE_<name>` env var](https://doc.rust-lang.org/cargo/reference/environment-variables.html) which is only set while building the binary or an integration test. I suspect the tests are falling through to the `insta-cmd` fallback path and reusing a prior (cached) binary which wouldn't be available in a clean `cargo test` run.

Reply:
Ah, that looks right, I deleted `target/debug/cargo-semver-checks` and it no longer works.  I think we can't use `insta_cmd` then and we have to make our own diff structure using `Check::check_release` (but that may be better in the long run with the speedups and new API).  Does that sound like a goal?